components/ILIAS/LegalDocuments/classes/Table/DocumentTable.php

```diff
@@ -20,149 +20,232 @@
 
 namespace ILIAS\LegalDocuments\Table;
 
-use ILIAS\LegalDocuments\Value\CriterionContent;
 use Closure;
-use ILIAS\LegalDocuments\TableConfig;
-use ILIAS\LegalDocuments\Table;
-use ILIAS\LegalDocuments\Value\Document;
-use ILIAS\LegalDocuments\Value\Criterion;
-use ILIAS\LegalDocuments\Repository\DocumentRepository;
-use ILIAS\LegalDocuments\ConsumerToolbox\UI;
-use ILIAS\UI\Component\Component;
-use ILIAS\LegalDocuments\TableSelection;
+use DateTimeImmutable;
+use Generator;
+use ilCalendarSettings;
+use ilCtrl;
+use ilCtrlInterface;
 use ilDatePresentation;
 use ilDateTime;
-use ilNumberInputGUI;
-use DateTimeImmutable;
-
-class DocumentTable implements Table
+use ILIAS\Data\Factory;
+use ILIAS\Data\URI;
+use ILIAS\LegalDocuments\ConsumerToolbox\UI;
+use ILIAS\LegalDocuments\EditLinks;
+use ILIAS\LegalDocuments\Repository\DocumentRepository;
+use ILIAS\LegalDocuments\Value\Criterion;
+use ILIAS\LegalDocuments\Value\Document;
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\Table\Action\Single;
+use ILIAS\UI\Component\Table\OrderingBinding;
+use ILIAS\UI\Component\Table\OrderingRowBuilder;
+use ILIAS\UI\Component\Table\Action\Multi;
+use ILIAS\UI\Component\Table\Ordering;
+use ILIAS\UI\Renderer;
+use ILIAS\UI\URLBuilder;
+use ILIAS\UI\URLBuilderToken;
+use ilObjUser;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class DocumentTable implements OrderingBinding
 {
-    /** @var Closure(class-string, ... $constructor_args): object<class-string> */
-    private readonly Closure $create;
-    /** @var Closure(DateTimeImmutable): string */
-    private readonly Closure $format_date;
+    public const CMD_EDIT_DOCUMENT = 'editDocument';
+    public const CMD_DELETE_DOCUMENT = 'deleteDocument';
+    public const CMD_DELETE_DOCUMENTS = 'deleteDocuments';
+    public const CMD_ADD_CRITERION = 'addCriterion';
+
+    private readonly ServerRequestInterface|RequestInterface $request;
+    private readonly Factory $data_factory;
+    private readonly ilCtrl|ilCtrlInterface $ctrl;
+    private readonly Ordering $table;
+    private readonly Renderer $ui_renderer;
+    private ilObjUser $user;
 
     /**
-     * @param Closure(CriterionContent): Component $criterion_as_component
-     * @param null|Closure(class-string, ... $constructor_args): object<class-string> $create
-     * @param Closure(DateTimeImmutable): string $format_date
+     * @var Document[]
      */
+    private array $records;
```

Comment:
Please drop this property.
This is only used once, but you can directly call `$this->repository->all()` in `getRows`. Additionally currently the documents are all fetched in the constructor, which may end in unnecessary database fetches.